### PR TITLE
feat(browser): add an option to take screenshots if the browser test fails

### LINKED ISSUE #5956: Take a screenshot if the test fails
### Clear and concise description of the problem

It would be nice to see the screenshot of the page if the test fails in CI or in headless mode, otherwise it is impossible to debug.

### Suggested solution

Store the screenshot of the page at the moment of the test failure.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by sheremet-va**
### Description

Fixes #5956

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/browser/src/client/tester/runner.ts [hunk 1]
```diff
@@ -53,6 +54,12 @@ export function createBrowserRunner(
       }
     }
 
+    onTaskFinished = async (task: Task) => {
+      if (this.config.browser.screenshotFailures && task.result?.state === 'fail') {
+        await page.screenshot()
```
Comment (userquin) [inline packages/browser/src/client/tester/runner.ts:59]: shouldn't check if the screenshot is available? what if running  using preview provider?

Comment (sheremet-va) [reply packages/browser/src/client/tester/runner.ts:59]: Good catch

Comment (sheremet-va) [reply packages/browser/src/client/tester/runner.ts:59]: Added a warning

**Review by userquin — APPROVED** [review]
nice :heart:

**Pull request merged**